components/results_table/commons/results_table.lua

```diff
@@ -101,8 +108,201 @@ function BaseResultsTable:getQueryType()
 	error('Invalid querytype "' .. (args.querytype or '') .. '"')
 end
 
+function BaseResultsTable:create()
+	local data = self.args.data or self:queryData()
+
+	if Table.isEmpty(data) then
+		return
+	end
+
+	table.sort(data, function(placement1, placement2) return placement1.date > placement2.date end)
+
+	if self.config.onlyAchievements then
+		self.data = {data}
+		return self
+	end
+
+	-- split placements into years for non achievements
+	local splitData = Array.groupBy(data, function(placementData)
+		return placementData.date:sub(1,4)
+	end)
+
+	-- Set the header
+	Table.iter.forEach(splitData, function(dataSet)
+		dataSet.header = dataSet[1].date:sub(1,4)
+	end)
+
+	self.data = Array.sortBy(splitData, function(val) return val.header end, function(val1, val2) return val1 > val2 end)
+
+	return self
+end
+
+function BaseResultsTable:queryData()
+	local data = mw.ext.LiquipediaDB.lpdb('placement', {
+		limit = self.config.limit,
+		order = self.config.sort .. ' ' .. self.config.order,
+		conditions = self:buildConditions(),
+	})
+
+	if type(data) ~= 'table' then
+		error(data)
+	end
+
+	return data
+end
+
+function BaseResultsTable:buildConditions()
+	if self.args.customConditions then
+		return self.args.customConditions
+	end
+
+	local conditions = self:buildBaseConditions()
+
+	if self.args.additionalConditions then
+		return conditions .. self.args.additionalConditions
+	end
+
+	return conditions
+end
+
+function BaseResultsTable:buildBaseConditions()
+	local args = self.args
+
+	local conditions = ConditionTree(BooleanOperator.all)
+		:add{self:buildOpponentConditions()}
+
+	if args.game then
+		conditions:add{ConditionNode(ColumnName('game'), Comparator.eq, args.game)}
+	end
+
+	local startDate = args.startdate or args.sdate
+	if startDate then
+		-- intentional > here to keep it as is in current modules
+		-- possibly change to >= later
+		conditions:add{ConditionNode(ColumnName('date'), Comparator.gt, startDate)}
+	end
+
+	local endDate = args.enddate or args.edate
+	if endDate then
+		-- intentional < here to keep it as is in current modules
+		-- possibly change to <= later
+		conditions:add{ConditionNode(ColumnName('date'), Comparator.lt, endDate)}
+	end
+
+	if args.placement then
+		conditions:add{ConditionNode(ColumnName('placement'), Comparator.eq, args.placement)}
+
+	-- todo: find a better solution once it is decided how to store awards data
+	elseif not Logic.readBool(args.ignorePlacement) then
+		conditions:add{ConditionNode(ColumnName('placement'), Comparator.neq, '')}
+	end
```

Comment:
Doyou want to update this to handle award with `mode` storage?

Reply:
oh yeah, good catch
i changed it in the sandbox already but forgot to copy it over
will copy it over when back on pc
(would have made it with the next part anyways)